thoth/common/logging.py

```diff
@@ -149,6 +149,28 @@ def _get_sentry_integrations() -> List[object]:
     return integrations
 
 
+def before_send_handler(event, hint):
+    """Filter the errors caught before sending to Sentry.
+
+    This function ignores the exceptions passed in as a environment variable in a comma separated manner.
+    """
+    ignored_exceptions = os.getenv('THOTH_SENTRY_IGNORE_EXCEPTION')
+    if not ignored_exceptions:
+        return event
+    exceptions_split = ignored_exceptions.split(',')
+    if 'exc_info' in hint:
+        exc_type, exc_value, tb = hint['exc_info']
+        for exception in exceptions_split:
+            if exception.strip() == exc_type.__name__:
```

Comment:
This logic can be done in `init_logging` and it can set up global variables so we unroll these operations out of the actual `before_send_handler` to speed up these error messages handling and fail on invalid configuration before the application starts:

```python
parts = exception.rsplit(maxsplit=1)
if len(parts) != 2:
    # error about wrong configuration supplied
exc_module, exc_name = parts

```

The core checking logic in `before_send` can then iterate over global variables as initialized in `init_logging` and perform something like this:

```python

if exc_type.__name__ == exc_name and exc_type.__module__ == exc_module:
    return None
```

WDYT?

Reply:
So sometimes hint doesn't have exc_info, in that case we fall back to log_record. I think it would be a good idea to have that logic in the before_send_handler, also makes it a little more modular.

Reply:
@fridex I have moved it to the init_logging method and modified the before_send_handler accordingly. Could you have a look?